### ISSUE TITLE
Firefox 55 

### ISSUE THREAD

**Issue by pierg75** [vimperator/vimperator-labs#772]
##### Issue type:

- Version compatibility

##### Version:
From latest git 

##### Description:
Vimperator doesn't seem to work on firefox 55. The addon is loaded when disabling the multiprocess requirement but it doesn't work at all (no command seem to activate the bottom bar)

##### Expected behavior:
Should work
##### Steps to reproduce:
Install latest vimperator in Firefox 55

**Comment by samrocketman** [comment]
When you say "latest git" can you provide the git hash you used?  Over time, "latest git" loses relevance.

**Comment by smilingsun** [comment]
Vimperator 3.16.0 does not work on Firefox 55.

**Comment by vp1981** [comment]
It (still?) works on my laptop: Archlinux x86_64 with
Mozilla/5.0 (X11; Linux x86_64; rv:55.0) Gecko/20100101 Firefox/55.0
vimperator: 3.16.0

**Comment by WuTheFWasThat** [comment]
for me, things mostly work.  however, buffer completion (b) no longer works for me!  it just shows "No completions".  

also, I saw "TypeError: tab.linkedbrowser.lastURI is null", at some point.  Not sure if related

**Comment by vp1981** [comment]
2WuTheFWasThat: on my system (Archlinux x86_64, Firefox from distribution, started and "managed" by systemd) the 'b' works fine: if I type TAB I can go through list of buffers, if I type the beginning (a number) and press TAB the corresponding buffer name is completed.

May be I'm lucky because the only issue I faced (long ago actually) is in searching: if I start search by pressing '/' it begins but don't close search input after I press RET but still has focus on it, and if I press RET second type I see:
```
TypeError: findbar._setHightlightTimeout is not a function
```
but search and highlighting seems to work (I can use 'n' to go to next match).

**Comment by roSievers** [comment]
@WuTheFWasThat Does (b) completion work for you after pressing it a few times and scrolling around a bunch? For me it doesn't work when I just restarted Firefox, but will start to work after a few tries. (Firefox from arch repo as well, using Antergos)

**Comment by hammerheadlemon** [comment]
Seeing problems in Firefox 55.0.1, vimperator 3.14.0. No way to access vimperator Preferences (button click does nothing). No `:buffers`, no `:tab`, no `f` to show links (`Reference Error: tab is not defined`). That's as far as I've got so far.

**Comment by vp1981** [comment]
Guys, you are scaring me..., I can't browsing effectively without vimperator.

I'll hope vimperator will continue to work fine on all my systems.

@hammerheadlemon, why you don't update vimperator to ver. 3.16.0?

**Comment by hammerheadlemon** [comment]
@vp1981 3.14.0 is default download seemingly and building and signing own version isn't working for me. Is there a way to install 3.16.0 without having to create your own keys, install npm, etc?

**Comment by vp1981** [comment]
@hammerheadlemon, that's funny. I installed vimperator using builtin mechanism, though I use firefox sync so may be this somehow help me to install ver. 3.16.0.

According to addons.mozilla.org, https://addons.mozilla.org/ru/firefox/addon/vimperator/versions/, ver. 3.16.0 is newer but is marked as not compatible with firefox 55.

I think ver. 3.14.0 was erroneously marked as compatible with firefox up to ver. 56, while ver. 3.16.0 was marked correctly.

I would try to remove and install vimperator again, may be this will help to install ver. 3.16.0

**Comment by hammerheadlemon** [comment]
@vp1981 Yeah, I'm seeing 3.16.0 marked as not compatible with my version of Firefox too. Interesting that the builtin mechanism gave you 3.16.0. I just tried reinstalling and still getting 3.14.0. I also use sync.

![image](https://user-images.githubusercontent.com/246857/29217652-574f31ca-7eaa-11e7-98ee-3adeaceb3464.png)

**Comment by WuTheFWasThat** [comment]
@roSievers pressing it a few times and scrolling around a bunch doesn't seem to help

stuff like "buffer! 2" does actually work.  but "buffers" does nothing

**Comment by SammysHP** [comment]
```
Vimperator:	3.16.0.20170426105746 (created: 2017/04/26 10:57:46)
Firefox:	Mozilla/5.0 (X11; Linux x86_64; rv:55.0) Gecko/20100101 Firefox/55.0
```

No problems so far.

**Comment by vp1981** [comment]
@hammerheadlemon, that very strange. Anyway, today I updated two Windows machines with Firefox 54 (I updated both Windows and Firefox). On one host there was no vimperator in Firefox (ver. 54) and Firefox itself couldn't update. I downloaded new installer, install new version of Firefox (55, i686) and installed vimperator after that using builtin mechanism, now this host has Firefox 55 and vimperator 3.16.0. Still not _all_ features are work well, for example 'd' doesn't work.

The other host also had Firefox 54 but this host is x86_64. Now I installed vimperator before the upgrade (again, Firefox wasn't able to download new version), upgraded Firefox to ver. 54, however I didn't test how vimperator works (will do it tomorrow).

On other host with (L)Ubuntu I have Firefox 54 (x86_64) and vimperator 3.16.0 and it works fine.

As I reported, vimperator 3.16.0 (basically) works fine on other my hosts with Linux (Archlinux x86_64).

According to SammysHP report I guess that this may be a Windows problem (actually not Windows itself but Firefox for Windows because either Ubuntu or Archlinux build Firefox from source).

**Comment by SammysHP** [comment]
I found a small (tiny) issue: The background color dependent SSL status icon does not work anymore because the bright version of the icon was removed in FF55.

**Comment by bs76** [comment]
After update to FF55 \:wopen does not seem to work for me

**Comment by dschrempf** [comment]
When I press `f`, I get a type error and nothing happens:

    TypeError: 'stopPrpagation' called on an object that does not implement inferface Event.

I am using the latest version on git, commit ae0df8040bc2821d6d0ce39a9c2f3b8feda4a3e4

**Comment by SammysHP** [comment]
@dschrempf 
Can you give us some more information? Does it happen on every page? Do you have other addons enabled?

**Comment by dschrempf** [comment]
I am using `waterfox 55.0.1`. When rolling back to `waterfox 54.0.1` everything works perfectly.

    Vimperator:	3.16.0 (created: 2017/08/17 10:24:18)
    Firefox:	Mozilla/5.0 (X11; Linux x86_64; rv:54.0) Gecko/20100101 Firefox/54.0.1 Waterfox/54.0.1

**Comment by sybix** [comment]
Hi,

I succed to install vimperator v3.16.0 on firefox 55, but uninstalling it, use a other browser to download the right version, and install it from file.

Maybe you can only uninstall vimperator and go to this link to force install the right version : https://addons.cdn.mozilla.net/user-media/addons/4891/vimperator-3.16.0-fx.xpi?filehash=sha256%3A64063090f48936ee7557908fb659e82ad7fe864f78f6f9b322bdfa95bfd1a497

**Comment by SammysHP** [comment]
We cannot maintain support for forks of Firefox. If it works in Firefox, but not in Watefox, this is not a problem of Vimperator but the changes in the Fork.

**Comment by dschrempf** [comment]
Hi, thank you. It works with Firefox 55. I had switched to Waterfox because I was not able to activate the (then unverified) Vimperator for a longer period of time a few months ago.

FYI: I tried Vimium because of these issues and concluded again that Vimperator is by far the better way to go. It would be a pity if the port to WebExtensions fails. Thanks for you work!

**Comment by davidsierradz** [comment]
If you have problems with Waterfox 55, search in about:config for 'browser.tabs.remote.autostart.2' and set to false, this seems to fix the issues so far.

**Comment by rsenna** [comment]
I tried to install Vimperator in a brand new Firefox, from Firefox Add-Ons page. I ended up with incompatible Vimperator version 3.14.0. Tried to use @sybix link: didn't work, since this is a brand new installation and xpinstall.signatures.required was set to the default value true. After setting it to false I was able to install version 3.16.0 and everything is working again.

**Comment by SammysHP** [comment]
`xpinstall.signatures.required` should not have any influence since a few versions. But for 3.16 max version is FF 51. We just need a new release.

**Comment by gyphub** [comment]
@SammysHP which commit corresponds to version 3.16.0.20170426105746? I built and installed the current HEAD (ae0df80) and it behaves strangely on FF 55.0.2. The `B` and `b` commands don't work for instance, and sometimes I have to right-click with my mouse to get any reaction at all.

**Comment by SammysHP** [comment]
HEAD should work fine, I was not able to reproduce the issue with <kbd>b</kbd>. How did you install the addon? Do you have compiled Firefox yourself?

**Comment by gyphub** [comment]
I checked out the repository, ran `make` and installed through the "Install Add-On From File..." command. This is the Firefox packaged with Ubuntu.
I hit `b`, type a string that would match a tab, hit Enter and get:
`TypeError: browser.contentDocument is undefined`

**Comment by dschrempf** [comment]
Did you reset your configuration by moving your `~/.mozilla/` out of the
way or deleting it before doing so?

**Comment by gyphub** [comment]
No, I've never done that.

However, it seems that the issues appear solely in windows that were restored from a browsing session from before the FF55 update. In new windows (created and restored after the update) everything works as it should.

**Comment by WuTheFWasThat** [comment]
yes @gyphub, you're right!  it seems I can use firefox 55 after all :) - for some reason I didn't think to test a new window

**Comment by dbolgheroni** [comment]
Did not know about Waterfox until this thread. Glad to see Vimperator (from master HEAD) works great with 55.0.2.

**Comment by ChShersh** [comment]
I've installed Vimperator 3.16 despite the fact that Add-ons Manager says it's incompatible with my Firefox 55 and it worked.

**Comment by pucerpocok** [comment]
@gyphub Thank you!
Based on your link I found the setting browser.sessionstore.restore_tabs_lazily.
After I set it to false and restarted the buffer completition works also in restored window.

**Comment by nmeum** [comment]
So…if HEAD works fine wouldn't it be time for a new release?

**Comment by bioe007** [comment]
Following @sybix comment, https://github.com/vimperator/vimperator-labs/issues/772#issuecomment-323297653 worked for me.
